libswdg/swdg.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Software watchdog library
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "swdg.h"
+
+#include <sys/threads.h>
+#include <sys/time.h>
+
+#if 0
+#include <stdio.h>
+#define DEBUG(fmt, ...) printf("swdg: " fmt, ##__VA_ARGS__)
+#else
+#define DEBUG(fmt, ...)
+#endif
+
+
+#ifndef SWDG_CHAN_CNT
+#define SWDG_CHAN_CNT 4
+#endif
+
+
+#ifndef SWDG_PRIORITY
+#define SWDG_PRIORITY 3
+#endif
+
```

Comment:
Well, I don't like defines in libs because it disallows to use it in multiple apps. Also for now we need to pass it in board_config which is not a good place for such defines.

Maybe malloc on init would be better pattern?

Reply:
Yes, as agreed, done